docs/knowledgebase/getting-started/index.md

```diff
@@ -74,119 +74,75 @@ rustup default stable
+project and different projects may use different mechanisms to communicate this version to
+developers. For instance, the Polkadot client specifies this information in its
+[release notes](https://github.com/paritytech/polkadot/releases). The Substrate Node Template uses
+an
+[init script](https://github.com/substrate-developer-hub/substrate-node-template/blob/master/scripts/init.sh)
+and
+[Makefile](https://github.com/substrate-developer-hub/substrate-node-template/blob/master/Makefile)
+to specify the Rust nightly version and encapsulate the following steps. Use Rustup to install the
+correct nightly:
 
 ```bash
 rustup install nightly-<yyyy-MM-dd>
 ```
 
----
-**NOTE**
-Due to a regression in the rust compiler, using the newest rust nightly for compiling
-the runtime will result in compilation errors. Therefore, it is advised to use the
-following version until this issue is resolved:
-
-```bash
-rustup install nightly-2020-10-06
-```
-
----
-
 #### Wasm Toolchain
 
-Now, configure the choosen nightly version to work with the Wasm compilation target:
+Now, configure the nightly version to work with the Wasm compilation target:
 
 ```bash
 rustup target add wasm32-unknown-unknown --toolchain nightly-<yyyy-MM-dd>
 ```
 
 #### Specifying Nightly Version
 
-When a Substrate based project builds its included runtime it picks the latest
-installed nightly version by default. If the nightly version is incompatible
-you can override that decision by setting the `WASM_BUILD_TOOLCHAIN` environment variable
-when building the project by using the following command:
+Use the `WASM_BUILD_TOOLCHAIN` environment variable to specify the Rust nightly version a Substrate
+project should use for Wasm compilation:
 
 ```bash
-WASM_BUILD_TOOLCHAIN=nightly-<yyyy-MM-dd> cargo build
+WASM_BUILD_TOOLCHAIN=nightly-<yyyy-MM-dd> cargo build --release
 ```
 
-Note that this builds only the runtime with the specified toolchain. The rest of project will
-be compiled with your default toolchain, which is usually the latest installed stable toolchain.
+#### Developing Substrate
 
-#### Latest Nightly
-
-If you want to build the runtime with the latest nightly compiler which should **generally** be
-possible you can install the unspecific `nightly` toolchain:
+Developers that are building Substrate _itself_ should always use the latest bug-free versions of
+Rust stable and nightly. This is because the Substrate codebase follows the tip of Rust nightly,
+which means that changes in Substrate often depend on upstream changes in the Rust nightly compiler.
+To ensure your Rust compiler is always up to date, you should run:
 
 ```bash
-rustup install nightly
+rustup update
+rustup update nightly
 rustup target add wasm32-unknown-unknown --toolchain nightly
 ```
 
-This toolchain is not tied to a specific version and will be updated just as the
-`stable` toolchain:
+**It may be necessary to occasionally rerun `rustup update`** if a change in the upstream Substrate
```

Comment:
Id remove this comment as again it creates more questions than answers and by the time people need to rerun `rustup update` I hope they are far enough into the community that this is a known req at times

Reply:
I think this is something that the core developers have asked that we keep in the docs in order to help with some common questions they receive. I hear what you're saying for sure and I do think it's important we don't get _too_ into the weeds in this doc, but given the above and the fact that this statement seems like it logically dovetails with the preceding statement I think I'm going to leave it at this time.